onnx/defs/nn/defs.cc

```diff
@@ -2832,4 +2832,100 @@ ONNX_OPERATOR_SET_SCHEMA(
               schema.BuildFunction(functionProto);
               return true;
             }));
+
+static const char* RotaryEmbedding_ver23_doc = R"DOC(
+RotaryEmbedding is the implementation of rotary positional embeddings (RoPE) based on the paper https://arxiv.org/pdf/2104.09864.
+The positions are represented as rotation matrices that are multiplied to query and key
+before the inner product of query and key is taken.
+
+Rotary embeddings are defined using the below functions:
+
+    def rotate_half(x):
+        """Rotates half the hidden dims of the input."""
+        x1 = x[..., : x.shape[-1] // 2]
+        x2 = x[..., x.shape[-1] // 2 :]
+        return torch.cat((-x2, x1), dim=-1)
+
+    def apply_rope(x, cos, sin, position_ids):
```

Comment:
Is this supposed to be RotaryEmbedding? If so, please use that name. It is unclear what is the relationship between `apply_rope` and `RotaryEmbedding`.

Reply:
Can you please clarify the shapes of the different arguments? It is hard to understand all the squeeze ops without understanding what the dimensions represent.